presentation-api/receiving-ua/common.js

```diff
@@ -7,4 +7,40 @@
     toController: '0c382524-5738-4df0-837d-4f53ea8addc2',
     toReceiver: 'a9618cd1-ca2b-4155-b7f6-630dce953c44'
   }
+
+  // handle a test result received from a receiving page
```

Comment:
What `parseValue` and `parseResult` gives us is great! One thing that I do not like is that these two functions will break if the format of the messages reported by the test harness changes slightly for some reason (e.g. because someone decides to clarify them further).

Could we make these functions more generic so that they cannot fail? Typically, what I have in mind is:
1. in `parseValue`: wrap the call to `JSON.parse` in a `try... catch` and report the value as-is if it cannot be parsed.
2. in `parseResult`: add a `default` case in the switch statement that calls `assert_false` for instance with the full message received.
3. in `parseResult`: if match fails, report the full message with an `assert_false` call as well.

In other words, my goal here would be to make sure the error gets reported no matter what. Now, calling `assert_false` might be confusing. It might make more sense to call `assert_unreached` with the message prefixed by something like "Unexpected failure message received from test harness: "